ldap-api/src/test/java/ca/bc/gov/hlth/ldapapi/controller/UsersControllerTest.java

```diff
@@ -0,0 +1,196 @@
+        );
+    }
+
+    @Test
+    public void postUser_unknownUser_emptyBody() {
+        Map<String, String> data = new HashMap<>();
+        data.put("userName", "not_a_real_user");
+        data.put("password", "not_a_real_password");
+        ResponseEntity<Map> response = restTemplate.postForEntity(urlUnderTest, data, Map.class);
+        assertTrue(response.getBody().isEmpty());
+    }
+
+    @Test
+    public void postUser_lockedUser_isLocked() {
+        Map<String, String> data = new HashMap<>();
+        data.put("userName", "2-primehcimintegrationtest");
+        // This is a valid password, but "authenticated" will still be false.
+        data.put("password", PASSWORD);
+        ResponseEntity<Map> response = restTemplate.postForEntity(urlUnderTest, data, Map.class);
+        assertAll(
+                () -> assertFalse((Boolean) response.getBody().get("authenticated")),
+                () -> assertFalse((Boolean) response.getBody().get("unlocked")),
+                () -> assertTrue(response.getBody().containsKey("userName")),
+                () -> assertFalse(response.getBody().containsKey("hcmuserrole")),
+                () -> assertEquals(3, response.getBody().get("remainingAttempts"))
+        );
+    }
+
+    @Test
+    public void postUser_invalidPasswordThreeTimes_lockedAccount() throws NamingException {
+        try {
+            Map<String, String> data = new HashMap<>();
+            data.put("userName", USERNAME);
+            data.put("password", "bad_password");
+
+            ResponseEntity<Map> response = restTemplate.postForEntity(urlUnderTest, data, Map.class);
+            assertAll(
+                    () -> assertFalse((Boolean) response.getBody().get("authenticated")),
+                    () -> assertTrue((Boolean) response.getBody().get("unlocked")),
+                    () -> assertEquals(2, response.getBody().get("remainingAttempts"))
+            );
+
+            ResponseEntity<Map> response2 = restTemplate.postForEntity(urlUnderTest, data, Map.class);
+            assertAll(
+                    () -> assertFalse((Boolean) response2.getBody().get("authenticated")),
+                    () -> assertTrue((Boolean) response2.getBody().get("unlocked")),
+                    () -> assertEquals(1, response2.getBody().get("remainingAttempts"))
+            );
+
+            ResponseEntity<Map> response3 = restTemplate.postForEntity(urlUnderTest, data, Map.class);
+            assertAll(
+                    () -> assertFalse((Boolean) response3.getBody().get("authenticated")),
+                    () -> assertFalse((Boolean) response3.getBody().get("unlocked")),
+                    () -> assertEquals(3, response3.getBody().get("remainingAttempts"))
+            );
+        } finally {
+            SearchResult searchResult = searchUser(USERNAME);
+            unlockUserAccount(searchResult.getName() + "," + LDAP_SEARCH_BASE);
+        }
+    }
+
```

Comment:
I'm thinking of refactoring `LdapService` to extract `lockUserAccount` and `searchUser` into their own class. It would have no dependencies on Spring and could be tested independently. Then the integration tests could call that class instead of duplicating the logic here.